templates/gcp/README.md

```diff
@@ -0,0 +1,248 @@
+H2O Open Source clusters on GCP
+===============================
+Terraform deployment template to spin up a 3 node H2O Open Source cluster in GCP. 
+
+This template is a work in progress and is provided without any warranty or support. You are free to refer/modify it as you need.
+
+There are two distinct parts to this setup
+1. Setting up the GCP project, service account, VPC, Subnet etc. Also in this step we create the GCP compute instance we call __Workspace__. All H2O users will need to ssh to this workspace instance to create the H2O cluster in the private subnet (not directly accessible). The workspace instance is in the public subnet an forms the gateway for all communication between the H2O cluster and machine of data scientist.
+2. Creating the H2O cluster using the `h2ocluster` tool available on the __Workspace__ instance
+
+Step 1: Main Infrastructure and Workspace Setup
+-----------------------------------------------
+
+These activities are performed by someone who have Cloud Admin privileges. In this step we perform pre-required activities using the browser on the GCP console, and then setup `gcloud` sdk tool on a machine where we use `terraform` to get the entire infrastructure setup.
+
+#### Create a GCP Project
+- Using a web browser, login to [GCP Console](https://console.cloud.google.com/)
+- Create a new Project
+- Ensure [Billing is enabled](https://cloud.google.com/billing/docs/how-to/modify-project?authuser=1) for the project
+- Enable needed APIs and services (link is on the top of project dashboard)
+- **!! NOTE !!** - For this work I reused an existing project with the name `steamwithdataproc`. The name does not directly relate to this work, but ignore that for now.
+
+#### Setup gcloud cli
+- Preferably setup the `gcloud` sdk on a linux based machine, ideally used by the Cloud system admin team to manage the cloud infrastructure.
+- Follow steps to [install Google Cloud SDK](https://cloud.google.com/sdk/docs/quickstart)
+- Create a new gcloud profile and authenticate
+	```
+	$ gcloud config configurations create hemen-h2oai
+	Created [hemen-h2oai].
+	Activated [hemen-h2oai].
+	$ gcloud auth login
+	Your browser has been opened to visit:
+	
+	    https://acco ..... deleted .... t_account
+	
+	You are now logged in as [hemen.kapadia@h2o.ai].
+	Your current project is [None].  You can change this setting by running:
+	  $ gcloud config set project PROJECT_ID
+	```
+
+- Setup project. You would have already created a project from the GUI as discussed earlier. Ensure it has billing enabled as well as services API enabled.
+	```
+	$ gcloud config set project steamwithdataproc
```

Comment:
Yes. That is one option. Since I was only working on this project I used the `gcloud config` route to set the values. That way I do not need to specify the project name in each command that would follow. Same strategy used for zone region etc.